src/pocketmine/event/HandlerList.php

```diff
@@ -43,19 +41,20 @@ class HandlerList{
 	 */
 	private static $allLists = [];
 
-	public static function bakeAll(){
-		foreach(self::$allLists as $h){
-			$h->bake();
-		}
-	}
+	/**
+	 * @var HandlerList[][] a 2D array of HandlerList.
+	 *                      The first dimension index is the class name, containing an array of HandlerList for that class and its superclasses.
+	 *                      The second dimension index is an integer >= 0, starting from the HandlerList for that class.
```

Comment:
I don't really understand why a multidimensional array is necessary here. This could be implemented with HandlerLists referencing the HandlerList of its parent.

Since we see further down that this list can be dynamically updated, I wonder why we don't just recursively iterate through handler list parents and call all appropriate handlers instead of using this thing. This would simplify the process of registering handler lists for new events and remove the need for this 2D array.

This 2D array only seems to be used for the purpose of accessing all the handler lists for an event linearly, which doesn't really make sense to me when it can be done recursively with less confusing code.